jdk/src/share/classes/sun/nio/cs/ext/ExtendedCharsets.java

```diff
@@ -1184,6 +1185,8 @@ protected void init() {
 
         String map = AccessController.doPrivileged(
             new GetPropertyAction("sun.nio.cs.map"));
+        boolean isGB18030_2000 = "2000".equals(AccessController.doPrivileged(
+                new GetPropertyAction("jdk.charset.GB18030")));
```

Comment:
More user friendly would be to test for "2000" or "2023" and to throw IAE for other values; but since later JDK versions don't do this either this is probably fine.

Reply:
I agree, but yes this is something that should go into trunk and be backported down. I'm already a little uneasy about fixing 8310947 in 8u first.